src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineViewModel.cs

```diff
@@ -303,17 +310,39 @@ private static List<GapTimeEntryBlock> GenerateGapTimeEntryBlocks(List<TimeEntry
                         VerticalOffset = lastTimeEntry.Bottom,
                         HorizontalOffset = 0
                     };
-                    if (block.Height > 10) // Don't display to small gaps not to obstruct the view
+                    if (block.Height >= TimelineConstants.MinGapTimeEntryHeight) // Don't display too small gaps not to obstruct the view
                         gaps.Add(block);
                 }
                 lastTimeEntry = lastTimeEntry == null || entry.Bottom > lastTimeEntry.Bottom
                     ? entry 
                     : lastTimeEntry;
             }
-            
+
             return gaps;
         }
 
+        private static GapTimeEntryBlock GenerateRunningGapBlock(IEnumerable<TimeEntryBlock> timeEntries, Toggl.TogglTimeEntryView? running,
+            double curTimeOffset, int selectedScaleMode, DateTime selectedDate)
+        {
+            var lastTimeEntryBottom = timeEntries.Any() ? timeEntries.Max(te => te.Bottom) : 0;
+            if (running == null && selectedDate.Date == DateTime.Today && !lastTimeEntryBottom.IsCloseTo(0) && curTimeOffset >= lastTimeEntryBottom + TimelineConstants.MinGapTimeEntryHeight)
+                return new GapTimeEntryBlock(
+                    (offset, height) =>

```

Comment:
GapTimeEntryBlock accepts `Func<double, double, string> addNewTimeEntry` as constructor parameter, where the first two doubles are supposed to be vertical offset and height. Here height is not used because we don't need the end time for the TE, but it's used for all other gap time entry blocks.

Reply:
👍